File: EcommerceSystem/Models/Category.cs
/*
 * File: Category.cs
 * Author: Athuluwage T.N
 * Description: This file contains the Category class, which represents a product category in the E-commerce system. It includes properties for the category's unique identifier, name, and a flag indicating whether the category is active.
 * Created: 08/10/2024
 */


using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace EcommerceSystem.Models
{
    public class Category
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string CategoryID { get; set; }

        public string Name { get; set; }

        public bool IsActive { get; set; } // Determines if the category is active
    }
}

File: EcomAndroidApp/app/src/main/java/com/example/e_comandroidapp/models/Category.java
package com.example.e_comandroidapp.models;

//Category model class
public class Category {

    private String categoryId;
    private String name;
    private boolean isActive;

    public Category(String categoryId, String name, boolean isActive) {
        this.categoryId = categoryId;
        this.name = name;
        this.isActive = isActive;
    }

    public String getCategoryId() {
        return categoryId;
    }

    public String getName() {
        return name;
    }

    public boolean isActive() {
        return isActive;
    }

    public void setCategoryId(String categoryId) {
        this.categoryId = categoryId;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setActive(boolean active) {
        isActive = active;
    }

    @Override
    public String toString() {
        return "Category{" +
                "categoryId='" + categoryId + '\'' +
                ", name='" + name + '\'' +
                ", isActive=" + isActive +
                '}';
    }
}
